interpreter/cling/lib/Interpreter/CIFactory.cpp

```diff
@@ -962,21 +1060,65 @@ static void stringifyPreprocSetting(PreprocessorOptions& PPOpts,
     // Set up the ASTContext
     CI->createASTContext();
 
-    if (OnlyLex) {
-      assert(customConsumer == nullptr && "Can't specify a custom consumer"
-                                          " when in OnlyLex mode");
-      class IgnoreConsumer: public clang::ASTConsumer {};
-      CI->setASTConsumer(
-          std::unique_ptr<clang::ASTConsumer>(new IgnoreConsumer()));
-    } else {
+    std::vector<std::unique_ptr<ASTConsumer>> Consumers;
+
+    if (!OnlyLex) {
       assert(customConsumer != nullptr && "Need to specify a custom consumer"
                                           " when not in OnlyLex mode");
-      CI->setASTConsumer(std::unique_ptr<clang::ASTConsumer>(customConsumer));
+      std::unique_ptr<clang::ASTConsumer> UniqueConsumer(customConsumer);
+      Consumers.push_back(std::move(UniqueConsumer));
     }
 
+    // With C++ modules, we now attach the consumers that will handle the
+    // generation of the PCM file itself.
+    if (COpts.CxxModules) {
+      // Code below from the (private) code in the GenerateModuleAction class.
+      llvm::SmallVector<char, 256> Output;
+      llvm::sys::path::append(Output, COpts.CachePath,
+                              COpts.ModuleName + ".pcm");
```

Comment:
Shouldn't we mark these ROOT-specific PCMs by giving them a more distinctive name?